src/unittest/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2022 Istituto Italiano di Tecnologia (IIT)
+# All rights reserved.
+# Author: Luca Tricerri
+# This software may be modified and distributed under the terms of the
+# BSD-3-Clause license. See the accompanying LICENSE file for details.
+
+cmake_minimum_required(VERSION 3.5)
+
+include(FetchContent)
+
+#GTEST
+FetchContent_Declare(
+        googletest
+        GIT_REPOSITORY "https://github.com/google/googletest"
+        GIT_TAG        main
```

Comment:
I would suggest to use a fixed tag (instead of a  branch) for any private dependency we include via FetchContent to avoid that tests are broken at some point also for old releases.

Reply:
Hi @triccyx 
Could you address this comment too?